### PR TITLE
RFC: evals api design spec

### PR DESCRIPTION
**Pull request by mikeldking**

### CODE REVIEW

File: tutorials/evals_ergonomics.ipynb [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    "prompt_template = ClassificationPromptTemplate(\n",
+    "    template=\"some prompt template the user is testing out with {context} and {question}\",\n",
+    "    rails=[\"relevant\", \"irrelevant\"],\n",
+    ")\n",
+    "clf = LLMClassifier(model=model, prompt_template=prompt_template)\n",
+    "trace_df[\"relevant\"] = clf.predict_dataframe(trace_df)\n",
+    "\n",
+    "# import back into phoenix\n",
+    "# accepts pandas dataframe or pandas series indexed with the same span IDs\n",
+    "session.import_evals(trace_df[\"relevant\"])"
```
Comment (mikeldking) [inline tutorials/evals_ergonomics.ipynb:61]: this is probably too many lines and too much manual steps - evals will have to be runnable on spans but the user shouldn't be forced to think about dataframes in this way. All they care about is the evals.

In addition we can't think of evals as singular things - it's really the derived metrics that will matter

File: src/phoenix/evals.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+# classification evaluator before computing the score
+
+
+# evals
+class Evals:
+    def __init__(self, evaluators: List[Evaluator], job_config: Optional[JobConfig] = None) -> None:
+        raise NotImplementedError()
+
+    @classmethod
+    def from_names(
```
Comment (mikeldking) [inline src/phoenix/evals.py:278]: not sure this path is needed - it foces two paths for construction

Comment (axiomofjoy) [reply src/phoenix/evals.py]: Removed.

File: tutorials/evals_ergonomics.ipynb (hunk 1 above)
Comment (axiomofjoy) [reply tutorials/evals_ergonomics.ipynb]: Overhauled this section. Simplified and created a function for running multiple evals at once.

**Pull request closed without merging**